tests/components/cover/test_command_line.py

```diff
@@ -44,41 +42,42 @@ def test_query_state_value(self):
 
     def test_state_value(self):
         """Test with state value."""
-        with tempfile.TemporaryDirectory() as tempdirname:
-            path = os.path.join(tempdirname, 'cover_status')
-            test_cover = {
-                'command_state': 'cat {}'.format(path),
-                'command_open': 'echo 1 > {}'.format(path),
-                'command_close': 'echo 1 > {}'.format(path),
-                'command_stop': 'echo 0 > {}'.format(path),
-                'value_template': '{{ value }}'
-            }
-            self.assertTrue(setup_component(self.hass, cover.DOMAIN, {
-                'cover': {
-                    'platform': 'command_line',
-                    'covers': {
-                        'test': test_cover
-                    }
+        test_cover = {
+            'command_state': 'cat cover_status',
+            'command_open': 'echo 1 > cover_status',
+            'command_close': 'echo 1 > cover_status',
+            'command_stop': 'echo 0 > cover_status',
+            'value_template': '{{ value }}'
+        }
+        self.assertTrue(setup_component(self.hass, cover.DOMAIN, {
+            'cover': {
+                'platform': 'command_line',
+                'covers': {
+                    'test': test_cover
                 }
-            }))
+            }
+        }))
 
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('unknown', state.state)
+        with mock.patch('subprocess.call') as mock_call:
+            mock_call.return_value = True
 
             cover.open_cover(self.hass, 'cover.test')
-            self.hass.block_till_done()
-
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('open', state.state)
-
             cover.close_cover(self.hass, 'cover.test')
-            self.hass.block_till_done()
-
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('open', state.state)
-
             cover.stop_cover(self.hass, 'cover.test')
             self.hass.block_till_done()
 
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('closed', state.state)
+            self.assertEqual(mock_call.call_count, 3)
+            self.assertEqual(
+                mock_call.call_args_list,
+                [
+                    mock.call(test_cover['command_open'], shell=True),
+                    mock.call(test_cover['command_close'], shell=True),
+                    mock.call(test_cover['command_stop'], shell=True)
+                ]
+            )
```

Comment:
In that case, it seems to me that `tests/components/cover/test_command_line.py` works great the way that it was originally.  In this case, if I mocked out the IO, I would need to also mock out the `hass.states.get`, which defeats the purpose, IMO.  Plus, the IO is so minimal, there's probably not much benefit to mocking out the IO in this case.  Do you agree?

If so, I'll just revert these changes and make a note in the original PR comment above that there are no changes needed in `tests/components/cover/test_command_line.py` (and, from what it looks like `tests/components/switch/test_command_line.py`, too).

Alternatively, if there is a way to accomplish this, then please let me know and I will make the necessary changes.